code-review/class.User.php

```diff
@@ -0,0 +1,316 @@
+				':token' => $token
+			       );
+		if($is_activated!=null && $type!=null){
+			$params[':is_activated'] = $is_activated;
+			$params[':type'] = $type;
+			$sql = "INSERT INTO users (username, full_name, email, password, joined, is_activated, type, token)";
+			$sql .= " VALUES (:username, :full_name, :email, :password, :joined, :is_activated, :type, :token)";
+		} else {
+			$sql = "INSERT INTO users (username, full_name, email, password, joined, token)";
+			$sql .= "VALUES (:username, :full_name, :email, :password, :joined, :token)";
+		}
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public static function addToken($username, $token){
+		global $db;
+		$sql = "UPDATE users SET token=:token WHERE username = :username";
+		$params = array(
+				':token' => $token,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public static function updatePassword($password, $username){
+		global $db;
+		$password = Utils::hash($password);
+		$sql="UPDATE users SET password=:password, token = 0 WHERE username = :username";
+		$params = array(
+				':password' => $password,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+
+	public static function updateLastVisit($username){
+		global $db;
+		$last_visit = date("Y-m-d H-i-s");
+		$sql = "UPDATE users SET last_visit = :last_visit WHERE username = :username";
+		$params = array(
+				':last_visit' => $last_visit,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+
+	public static function getNumberOfUsers($search=null,$category=null) {
+		global $db;
+		if ($search != null && $category != null) {
+			$params[':search_string'] = '%'.$search.'%';
+			switch ($category) {
+				case "username":
+					$sql = "SELECT COUNT(*) as num FROM users WHERE username LIKE :search_string";
+					break;
+				case "full_name":
+					$sql = "SELECT COUNT(*) as num FROM users WHERE full_name LIKE :search_string";
+					break;
+				case "email":
+					$sql = "SELECT COUNT(*) as num FROM users WHERE email LIKE :search_string";
+					break;
+
+			}
+			$query = $db->query($sql,$params);
+		} else {
+			$sql = "SELECT COUNT(*) as num FROM users";
+			$query = $db->query($sql);
+		}
+
+		$result = $db->fetchArray($query);
+		return $result['num'];
+	}
+
+	public static function getNUsers ($start, $limit, $search=null, $category=null) {
+		global $db;
+		$params = array(
+				':start' => $start,
+				':limit' => $limit
+			       );
+		if ($search != null && $category != null) {
+			$params[':search_string'] = '%'.$search.'%';
+			switch ($category) {
```

Comment:
### Pažeidžiamos taisyklės
- [ ] Nenaudoti switch sakinių be default dalies
### Komentarai
Reikėtų aprašyti `default case`, klaidoms valdyti